wgpu-core/src/instance.rs

```diff
@@ -274,15 +303,31 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         ))]
         let mut adapters_vk = match instance.vulkan {
             Some(ref inst) if id_vulkan.is_some() => {
-                let adapters = inst.enumerate_adapters();
+                let mut adapters = inst.enumerate_adapters();
+                if let Some(surface) = compatible_surface {
```

Comment:
We should probably try to de-duplicate these with a function or macro per adapter type at some point if we continue to change these every once in a while :)

Reply:
yeah :) I honestly try to avoid macros, but I would welcome a PR if it does the job done